### DISCUSSION TITLE
Embedded mp4 videos don't work with Firefox

### DISCUSSION THREAD

**Discussion by lorenzo-rovigatti** [jupyter-book/mystmd#1749] [Q&A] [answered]
Hi! I'm trying to embed an mp4 in my book. The compilation works just fine, but the video shows up as frozen image when the page is opened with Firefox, while it works well with Chromium.

By the way, being able to have controls as suggested in #532 would be very useful!

**Comment by agoose77** [comment]
Could you report your version of Firefox and MyST? On my machine, I can successfully view an auto-playing video in 131.0.2 with MyST v1.3.12

**Comment by lorenzo-rovigatti** [comment]
Thanks for the prompt answer, and sorry for the lack of details in the original post. I'm using MyST v1.3.12 and FF 131.0.2 (basically the same versions as you).
I'm trying to test a bit more what happens, and I see a weird behaviour: as I refresh the page, the video is frozen, but if I scroll down or up enough and then come back, sometimes the video will start! You can try it yourself here: https://lorenzo-rovigatti.github.io/comp_bio_notes/folding#the-structure-module

**Comment by agoose77** [comment]
Hmm, I can't seem to replicate this with your site. I wonder if this is a spurious issue rather than something we can influence! 🤔

**Comment by lorenzo-rovigatti** [comment]
Here is how I can reproduce it 100% of the times on my (three) linux machines:

* Go to https://lorenzo-rovigatti.github.io/comp_bio_notes/folding#the-structure-module
* Wait till the page has loaded: the video will start
* Refresh the page: the video is frozen

However, note that I have also tried it on an Apple computer and I cannot reproduce the issue, so maybe it depends on my configuration...

**Comment by rowanc1** [comment]
There are some settings that stop a video from auto playing, for example if it isn't muted. We do do that though!

I have also seen that videos can be unplayable in Firefox because of the code used. Doesn't seem to be the case here though.

**Comment by agoose77** [comment]
I'm going to close this for now as we can't seem to reproduce it. Please do re-open if you find a way to reproduce it on e.g. another machine!

**Comment by rowanc1** [comment, marked as answer]
@lorenzo-rovigatti  if an MP4 video isn't playing in Firefox, I strongly suspect that the codec used in the video is not supported by the browser (like a newer version of H.265 mp4 codec), which could be due to an outdated codec or a codec that Firefox simply doesn't recognize; to fix this, you might need to re-encode the video using a compatible codec like H.264, check for Firefox updates, or investigate if any extensions are interfering with playback. 

Hopefully that helps.

**Comment by lorenzo-rovigatti** [comment]
Thank you, I'll give it a try!
